lib/pa11y.js

```diff
@@ -57,26 +40,60 @@ async function pa11y(url, options = {}, callback) {
 			options.timeout,
 			`Pa11y timed out (${options.timeout}ms)`
 		);
-
 		// Run callback if present, and resolve with results
-		if (callback) {
-			return callback(null, results);
-		}
-		return results;
-
+		return callback ? callback(null, results) : results;
```

Comment:
If the `results` and this line are outside the `try/catch` then we can ensure that `stateCleanup` is called before the callback is fired.
Which would eliminate the need for the `setTimeout` in the tests, which could cause flakey tests on slower running machines.